src/lib/isLicensePlate.js

```diff
@@ -4,6 +4,7 @@ const validators = {
   'de-DE': str =>
     /^((AW|UL|AK|GA|AÖ|LF|AZ|AM|AS|ZE|AN|AB|A|KG|KH|BA|EW|BZ|HY|KM|BT|HP|B|BC|BI|BO|FN|TT|ÜB|BN|AH|BS|FR|HB|ZZ|BB|BK|BÖ|OC|OK|CW|CE|C|CO|LH|CB|KW|LC|LN|DA|DI|DE|DH|SY|NÖ|DO|DD|DU|DN|D|EI|EA|EE|FI|EM|EL|EN|PF|ED|EF|ER|AU|ZP|E|ES|NT|EU|FL|FO|FT|FF|F|FS|FD|FÜ|GE|G|GI|GF|GS|ZR|GG|GP|GR|NY|ZI|GÖ|GZ|GT|HA|HH|HM|HU|WL|HZ|WR|RN|HK|HD|HN|HS|GK|HE|HF|RZ|HI|HG|HO|HX|IK|IL|IN|J|JL|KL|KA|KS|KF|KE|KI|KT|KO|KN|KR|KC|KU|K|LD|LL|LA|L|OP|LM|LI|LB|LU|LÖ|HL|LG|MD|GN|MZ|MA|ML|MR|MY|AT|DM|MC|NZ|RM|RG|MM|ME|MB|MI|FG|DL|HC|MW|RL|MK|MG|MÜ|WS|MH|M|MS|NU|NB|ND|NM|NK|NW|NR|NI|NF|DZ|EB|OZ|TG|TO|N|OA|GM|OB|CA|EH|FW|OF|OL|OE|OG|BH|LR|OS|AA|GD|OH|KY|NP|WK|PB|PA|PE|PI|PS|P|PM|PR|RA|RV|RE|R|H|SB|WN|RS|RD|RT|BM|NE|GV|RP|SU|GL|RO|GÜ|RH|EG|RW|PN|SK|MQ|RU|SZ|RI|SL|SM|SC|HR|FZ|VS|SW|SN|CR|SE|SI|SO|LP|SG|NH|SP|IZ|ST|BF|TE|HV|OD|SR|S|AC|DW|ZW|TF|TS|TR|TÜ|UM|PZ|TP|UE|UN|UH|MN|KK|VB|V|AE|PL|RC|VG|GW|PW|VR|VK|KB|WA|WT|BE|WM|WE|AP|MO|WW|FB|WZ|WI|WB|JE|WF|WO|W|WÜ|BL|Z|GC)[- ]?[A-Z]{1,2}[- ]?\d{1,4}|(AIC|FDB|ABG|SLN|SAW|KLZ|BUL|ESB|NAB|SUL|WST|ABI|AZE|BTF|KÖT|DKB|FEU|ROT|ALZ|SMÜ|WER|AUR|NOR|DÜW|BRK|HAB|TÖL|WOR|BAD|BAR|BER|BIW|EBS|KEM|MÜB|PEG|BGL|BGD|REI|WIL|BKS|BIR|WAT|BOR|BOH|BOT|BRB|BLK|HHM|NEB|NMB|WSF|LEO|HDL|WMS|WZL|BÜS|CHA|KÖZ|ROD|WÜM|CLP|NEC|COC|ZEL|COE|CUX|DAH|LDS|DEG|DEL|RSL|DLG|DGF|LAN|HEI|MED|DON|KIB|ROK|JÜL|MON|SLE|EBE|EIC|HIG|WBS|BIT|PRÜ|LIB|EMD|WIT|ERH|HÖS|ERZ|ANA|ASZ|MAB|MEK|STL|SZB|FDS|HCH|HOR|WOL|FRG|GRA|WOS|FRI|FFB|GAP|GER|BRL|CLZ|GTH|NOH|HGW|GRZ|LÖB|NOL|WSW|DUD|HMÜ|OHA|KRU|HAL|HAM|HBS|QLB|HVL|NAU|HAS|EBN|GEO|HOH|HDH|ERK|HER|WAN|HEF|ROF|HBN|ALF|HSK|USI|NAI|REH|SAN|KÜN|ÖHR|HOL|WAR|ARN|BRG|GNT|HOG|WOH|KEH|MAI|PAR|RID|ROL|KLE|GEL|KUS|KYF|ART|SDH|LDK|DIL|MAL|VIB|LER|BNA|GHA|GRM|MTL|WUR|LEV|LIF|STE|WEL|LIP|VAI|LUP|HGN|LBZ|LWL|PCH|STB|DAN|MKK|SLÜ|MSP|TBB|MGH|MTK|BIN|MSH|EIL|HET|SGH|BID|MYK|MSE|MST|MÜR|WRN|MEI|GRH|RIE|MZG|MIL|OBB|BED|FLÖ|MOL|FRW|SEE|SRB|AIB|MOS|BCH|ILL|SOB|NMS|NEA|SEF|UFF|NEW|VOH|NDH|TDO|NWM|GDB|GVM|WIS|NOM|EIN|GAN|LAU|HEB|OHV|OSL|SFB|ERB|LOS|BSK|KEL|BSB|MEL|WTL|OAL|FÜS|MOD|OHZ|OPR|BÜR|PAF|PLÖ|CAS|GLA|REG|VIT|ECK|SIM|GOA|EMS|DIZ|GOH|RÜD|SWA|NES|KÖN|MET|LRO|BÜZ|DBR|ROS|TET|HRO|ROW|BRV|HIP|PAN|GRI|SHK|EIS|SRO|SOK|LBS|SCZ|MER|QFT|SLF|SLS|HOM|SLK|ASL|BBG|SBK|SFT|SHG|MGN|MEG|ZIG|SAD|NEN|OVI|SHA|BLB|SIG|SON|SPN|FOR|GUB|SPB|IGB|WND|STD|STA|SDL|OBG|HST|BOG|SHL|PIR|FTL|SEB|SÖM|SÜW|TIR|SAB|TUT|ANG|SDT|LÜN|LSZ|MHL|VEC|VER|VIE|OVL|ANK|OVP|SBG|UEM|UER|WLG|GMN|NVP|RDG|RÜG|DAU|FKB|WAF|WAK|SLZ|WEN|SOG|APD|WUG|GUN|ESW|WIZ|WES|DIN|BRA|BÜD|WHV|HWI|GHC|WTM|WOB|WUN|MAK|SEL|OCH|HOT|WDA)[- ]?(([A-Z][- ]?\d{1,4})|([A-Z]{2}[- ]?\d{1,3})))[- ]?(E|H)?$/.test(str),
   'de-LI': str => /^FL[- ]?\d{1,5}[UZ]?$/.test(str),
+  'it-IT': str => /^[ABCDEFGHJKLMNPRSTVWXYZ]{2}\s?(?:\d{5}|\d{3}\s?[ABCDEFGHJKLMNPRSTVWXYZ]{2})$/.test(str),

```

Comment:
IMHO this REGEXP is really to simple.

As a result, many cases are not covered:

- Old plates with the 2 first letters standing for a city. Eg, for Florence (Firenze) the plate woulld be like `FI 12345`
- Old plates from the capital. Indeed they have a special form, like `ROMA 12345`

For a better covering, I would suggest a regexp of the form: `/^(<old-plate>|<new-plate>)$/`

with the following sub-expressions:

- old-plate: a list of all possible city codes, followed by 5 numbers : `(ROMA|AG|AL|AN|AO...)\s?\d{5}` **<sup>(1)</sup>**
- new-plate: `[ABCDEFGHJKLMNPRSTVWXYZ]{2}\s?\d{3}\s?[ABCDEFGHJKLMNPRSTVWXYZ]{2}`

**(1)** The full list can be found here: https://en.wikipedia.org/wiki/Vehicle_registration_plates_of_Italy#Province_codes_1927_to_present_day

Reply:
I forgot to mention that this PR add support only for car and motorcycle from ['94](https://en.wikipedia.org/wiki/Vehicle_registration_plates_of_Italy#1994-present), because I thought it was enough.

I didn't want to include old plates because there are a lot of format and exceptions for the most populous provinces (check the [Italian version of wikipedia](https://it.wikipedia.org/wiki/Targhe_d%27immatricolazione_dell%27Italia) that is more complete than the english one).

Old plates can be: AA 123456, but also 123456 AA and AA B12345. In the province of Torino the lettere was pushed to the bottom like AA 12345B, in the province of Rome and Milan the letter was placed in second position like AA 1A1234 and so on.
https://it.wikipedia.org/wiki/Targhe_d%27immatricolazione_dell%27Italia#Sistema%20di%20numerazione

IMHO to many exceptions that can raise false positives.

Reply:
I agree with @fasenderos, since there are too many edge cases to support and the game is not worth the candle.

Reply:
OK, at first sight it seemed to me as if the regex was intended to support those old plate numbers... but maybe I had it all wrong... 

So if old plates are not supported, this means the regex could be simplified to : 

```
/^[ABCDEFGHJKLMNPRSTVWXYZ]{2}\s?\d{3}\s?[ABCDEFGHJKLMNPRSTVWXYZ]{2}$/
```

what do you think ? :thinking:

Reply:
> OK, at first sight it seemed to me as if the regex was intended to support those old plate numbers... but maybe I had it all wrong...
> 
> So if old plates are not supported, this means the regex could be simplified to :
> 
> ```
> /^[ABCDEFGHJKLMNPRSTVWXYZ]{2}\s?\d{3}\s?[ABCDEFGHJKLMNPRSTVWXYZ]{2}$/
> ```
> 
> what do you think ? 🤔

This regex support car only. My regex works with [motorcycle too](https://en.wikipedia.org/wiki/Vehicle_registration_plates_of_Italy#Motorcycle_plates)

Reply:
OK, my bad :stuck_out_tongue_closed_eyes:

Reply:
@yannoff you're not totally wrong. I reread the motorcycle's plates specifications, 2 letters and 5 digits **BUT** for the 2 letters the _province codes are not used to avoid confusion (i.e. the plate after AF 99999 is AH 00000, because AG is the old provincial code for Agrigento)_.

So we have 3 choises:
- Use your regex that supports only car after '94;
- Exclude from my regex the old provincial codes ONLY for motorcycle's plate;
- Leave the regex as is, because a plate with the old provincial code and 5 digits i.e. AG 12345 can be an old car plates before '94.

Now I'm really confused :(

Reply:
Abbiamo pazienza :wink:

I don't know well how the library is intended to work *(are motorcycles included in other validators?)*, but I would opt for the 1st option - maybe with a little modifications to support the weird old-plates you mentioned above *(be it a little too permissive, we don't care)* - because :
- a few (but there are) cars with old plates are still in use in Italy
- the 2nd option would mean a way more complicated regex IMHO - without resolving the "unsupported old plates" issue 
- the 3rd is not the good solution, because some old plates might appear valid whereas others not - in other words, that means unexpected behavior for end users

Whatever the choice, a **caveat** would be welcome to document the unsupported plates (be it motorcycles, older than '94 or whatever)

Reply:
@yannoff I'm agree with you for 2nd and 3rd option, while for the 1st option, if it was my app I would prefer the simple regex, without partial supports for oldest plates, because IMHO:

- with `/^[ABCDEFGHJKLMNPRSTVWXYZ]{2}\s?\d{3}\s?[ABCDEFGHJKLMNPRSTVWXYZ]{2}$/` we can say "we support all car plates from '94 without exceptions';
- with ` /^(<old-plate>|<new-plate>)$/` we can say "we support all car plates from '94 without exceptions, while for the oldest one we supports only the really oldest plates, that doesn't have a letter in the place of one digit". 
- You're right when you say that a few old plates are still in use in Italy, but the weird old-plates (not supported by regex) that have a letter in place of one digit e.g. MI 1B2345 is newst than MI 123456, and most likely the one in use.

Anyway if it's ok for you this tradeoff, I'm go ahead with  `/^(<old-plate>|<new-plate>)$/`

Reply:
We can support this 😂

![01F7C3A4-6476-4A20-8B8B-AB001A27E731](https://user-images.githubusercontent.com/1219087/120902130-f5db6400-c63e-11eb-9ccd-97161c2effe1.jpeg)

Reply:
Regex updated with support for old italian plates https://github.com/validatorjs/validator.js/pull/1677/commits/68230c2a084a9032b4a8c94bc25e55c007be6df5